### PR TITLE
Add jsonSerialized output field

### LINKED ISSUE #1849: [Cosmos] Return ready-to-broadcast json for protobuf mode
Similar to current `json` of `SigningOutput`

```json
{
  "tx_bytes": "<output.serialized>",
  "mode": "BROADCAST_MODE_SYNC"
}
```

Related to #1818

### PR DESCRIPTION
**Pull request by catenocrypt**
## Description

Add json_serialized output field, directly usable as broadcast message.  
Rather straightforward adding of a new field.
Fixes #1849 .

## Testing instructions

Cosmos unit tests.

## Types of changes

* New feature (non-breaking change which adds functionality)

## Checklist

- [x] Prefix PR title with `[WIP]` if necessary.
- [x] Add tests to cover changes as needed.
- [x] Update documentation as needed.
- [x] I have read the [guidelines](https://developer.trustwallet.com/wallet-core/newblockchain#integration-criteria) for adding a new blockchain
- [x] If there is a related Issue, mention it in the description (e.g. Fixes #<issue_number> ).

### CODE REVIEW

File: src/proto/Cosmos.proto [hunk 1]
```diff
@@ -135,6 +135,9 @@ message SigningOutput {
     // Signed transaction in protobuf encoded form, Base64-encoded (Stargate case)
     string serialized = 3;
 
+    // Json format directly suitable for broadcast message, containing Stargate serialized field.
+    string json_serialized = 4;
```
Comment (vikmeup) [inline src/proto/Cosmos.proto:139]: I think in all our cases we always return data of transaction to be broadcasted to the network, unless there is other reasons to keep both?

**Review by vikmeup — APPROVED** [review]

File: src/Cosmos/Signer.cpp [hunk 2]
```diff
@@ -53,8 +53,11 @@ Proto::SigningOutput Signer::signProtobuf(const Proto::SigningInput& input) noex
         auto serializedTxRaw = buildProtoTxRaw(input, serializedTxBody, serializedAuthInfo, signature);
 
         auto output = Proto::SigningOutput();
-        output.set_serialized(Base64::encode(TW::data(serializedTxRaw)));
+        const string serializedBase64 =Base64::encode(TW::data(serializedTxRaw)); 
+        output.set_serialized(serializedBase64);
         output.set_signature(signature.data(), signature.size());
+        const string jsonSerialized = "{\"tx_bytes\": \"" + serializedBase64 + "\", \"mode\": \"BROADCAST_MODE_SYNC\"}";
```
Comment (hewigovens) [inline src/Cosmos/Signer.cpp:59]: `mode` should not be hardcoded, it needs to read from `proto` and set according value (https://docs.cosmos.network/master/core/proto-docs.html#broadcastmode)

File: src/proto/Cosmos.proto (hunk 1 above)
Comment (hewigovens) [reply src/proto/Cosmos.proto:139]: yeah, just replace the original `serialized` field, should be fine to change this

Comment (catenocrypt) [reply src/proto/Cosmos.proto:139]: OK, just replacing contents of the serialized field, but keeping its name.

**Review by hewigovens — APPROVED** [review]

**Pull request merged**